flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1952,9 +1952,11 @@ class ScalarExprLowering {
     mlir::Value retValBase = fir::getBase(retVal);
     if (fir::conformsWithPassByRef(retValBase.getType()))
       return retVal;
-    auto mem = builder.create<fir::AllocaOp>(getLoc(), retValBase.getType());
-    builder.create<fir::StoreOp>(getLoc(), retValBase, mem);
-    return fir::substBase(retVal, mem.getResult());
+    if (retVal.rank() != 0)
+      fir::emitFatalError(getLoc(),
+                          "internal: array function result not in memory");
```

Comment:
Do we really need "internal: " prefix? We are not consistent on the whole.

Note that emitFatalError() prints out messages in quite a different format than the front end's printf style error messages with the "^^^^^" marking, etc.

Reply:
Agreed, I was actually trying to be consistent with the message 10 lines above (https://github.com/flang-compiler/f18-llvm-project/blob/85caaf9309abea9487159754b468fb602494e820/flang/lib/Lower/ConvertExpr.cpp#L1948).
I have removeed the "internal: " from the one above. The error where this comment lies is removed since the code is calling `placeScalarValueInMemory` directly now.

Reply:
OK. No worries.